src/server.py

```diff
@@ -373,17 +385,50 @@ def play_sound(self, sound_event_id, connection, sender, path, iface,
                 self._DBUS_UNKNOWN_SOUND_EVENT_ID,
                 "sound event with id %s does not exist" % sound_event_id)
             return
-        self._cancel_countdown()
-        self.hold()
 
-        uuid_ = str(uuid.uuid4())
-        metadata = self.metadata[sound_event_id]
-        self.players[uuid_] = HackSoundPlayer(uuid_, metadata, sender, options)
-        self.players[uuid_].connect("eos", self.__player_eos_cb, uuid_)
-        self.players[uuid_].connect("error", self.__player_error_cb, uuid_,
-                                    connection, path, iface)
-        self.players[uuid_].play()
-        invocation.return_value(GLib.Variant('(s)', (uuid_, )))
+        # Insert players organized by sound event id.
+        if sound_event_id not in self._uuid_by_event_id:
+            self._uuid_by_event_id[sound_event_id] = set()
+
+        overlap_behavior =\
+            self.metadata[sound_event_id].get("overlap-behavior", "overlap")
+
+        uuid_ = None
+        if len(self._uuid_by_event_id[sound_event_id]) > 0:
+            tmp_uuid = next(iter(self._uuid_by_event_id[sound_event_id]))
```

Comment:
Since `set` is unordered, this won't return necessarily the first item added to it. On the other hand, why are you trying to get the first item, shouldn't be the latest one that you want to reuse?

Reply:
If I understand the PR correctly, only one UUID is stored in the set for the cases where checking UUID is relevant: restart and ignore. The default case "overlap" doesn't need a check.

Another option is, not store uuids in `self._uuid_by_event_id` for "overlap". And store one UUID (not a set) for the other two cases.

Reply:
Yes, there is actually just one ID. I didn't care about the order. But I will just use the UUID only for sounds with the "overlap-behavior".